tests/S3/S3ClientTest.php

```diff
@@ -2228,4 +2228,47 @@ public function testAddsForwardSlashIfEmptyPathAndQuery()
         );
         $s3->execute($command);
     }
+
+    public function testAutomaticallyComputesMD5ForPutObject()
+    {
+        $s3 = $this->getTestClient('s3');
+        $this->addMockResults($s3, [[]]);
+        $command = $s3->getCommand('PutObject',
+            ['Bucket' => 'foo', 'Key' => 'foo', 'Body' => 'test', 'AddContentMD5' => true]
+        );
+        $command->getHandlerList()->appendSign(
+            Middleware::tap(function ($cmd, $req) {
+                $this->assertSame(
+                    'CY9rzUYh03PK3k6DJie09g==',
+                    $req->getHeader('Content-MD5')[0]
+                );
+            })
+        );
+        $s3->execute($command);
+    }
+
+    public function testAutomaticallyComputesMD5ForUploadPart()
```

Comment:
Please add a test for this that is the inverse that checks that it's not added when AddContentMD5 is empty